### ISSUE TITLE
Proposal: Grammar: Computed property keys 

### ISSUE THREAD

**Issue by MegaUITeam** [jashkenas/coffeescript#4622]
Like this:
---
```javasript
w = 'prop'
q = {[w]:2}
```

There is no ambiguity, since after arrays `:` is never put.

I know, we can write like this:
```javascript
q = {"#{w}":2}
```
But it's not cool. In addition, we can not use link to Symbol, with the second use case
```Uncaught TypeError: Cannot convert a Symbol value to a string```

Besides
---
this is a correct javascript code, and compiles with Babel
```javascript
w = Symbol()
q = {[w]:2}
```

**Comment by microdou** [comment]
It's ES6 **Computed Property Keys**, previously discussed in #4338.

**Comment by GeoffreyBooth** [comment]
Closed via #4803.
